src/webview-js/toc-editor/toc-editor.jsx

```diff
@@ -0,0 +1,155 @@
+
+  const handleChange = (newChildren) => {
+    const { treesData, selectionIndices } = vscode.getState()
+
+    const newData = { ...data, ...{children: newChildren} }
+    const oldStructure = JSON.stringify(removeExpanded(data.children))
+    const newStructure = JSON.stringify(removeExpanded(newChildren))
+
+    if (data.children.length - newChildren.length > 3) {
+      // There's a bug that deletes the whole tree except for one element.
+      // Prevent this by not allowing high magnitude deletions
+      return
+    }
+
+    treesData[modifiesStateName][props.index].children = newChildren
+    vscode.setState({ treesData, selectionIndices })
+    setData(newData)
+
+    if (oldStructure !== newStructure) {
+      vscode.postMessage({treeData: newData})
+    }
+  }
+
+  const getNodeProps = ({ node }) => {
+    const typeToColor = {
+      'subcollection': 'green',
+      'module': 'purple'
+    }
+    return {
+      style: {
+        boxShadow: `0 0 0 4px ${typeToColor[node.type]}`,
+      },
+      title: node.title + (node.moduleid ? ` (${node.moduleid})` : '')
+    }
+  }
+
+  const canDrop = ({ nextParent}) => {
+    if (nextParent && nextParent.type === 'module') {
+      return false
+    }
+    return true
+  }
+  
+  return (
+    <div style={{ height: '100%' }}>
+      <SortableTree
+        treeData={data.children}
+        onChange={props.editable ? handleChange : () => {}} // non-editable treeData will never change
+        generateNodeProps={getNodeProps}
+        canDrop={props.editable ? canDrop : () => true} // Dropping item into non-editable trees will destroy the item
+        shouldCopyOnOutsideDrop={!props.editable}
+        dndType={nodeType}
+      />
+    </div>
+  )
+}
+
+const EditorPanel = (props) => {
+  const modifiesStateName = props.modifiesStateName
+  const trees = props.treesData
+  const [selection, setSelection] = useState(props.selectionIndex);
+
+  const handleSelect = (event) => {
+    const { treesData, selectionIndices } = vscode.getState()
+    const newSelection = parseInt(event.target.value)
+    selectionIndices[modifiesStateName] = newSelection
+    vscode.setState({ treesData, selectionIndices })
+    setSelection(newSelection)
+  }
+
+  return (
+    <div style={{display: 'flex', flexDirection: 'column', height: '99vh', width: '49vw'}}>
+      <select value={selection} style={{margin: '1rem', maxWidth: '300px'}} onChange={handleSelect}>
+        {trees.map((tree, i) => {
+          return (<option key={i} value={i}>{tree.title}</option>)
+        })}
+      </select>
+      <div style={{flexGrow: '1'}}>
+        {trees.map((tree, i) => {
+          return (
+            <div key={i} style={{height: '100%', display: selection === i ? 'block' : 'none'}}>
+              <ContentTree modifiesStateName={modifiesStateName} index={i} data={tree} editable={props.editable}/>
+            </div>
+          )
+        })}
+      </div>
+    </div>
+  )
+}
+
+const App = (props) => {
+  return (
```

Comment:
This `return` could be removed too unless it's intentionally kept here for legibility